src/t_zset.c

```diff
@@ -1186,23 +1186,23 @@ void zsetConvert(robj *zobj, int encoding) {
         zs->zsl = zslCreate();
 
         eptr = lpSeek(zl,0);
-        serverAssertWithInfo(NULL,zobj,eptr != NULL);
-        sptr = lpNext(zl,eptr);
-        serverAssertWithInfo(NULL,zobj,sptr != NULL);
-
-        while (eptr != NULL) {
-            score = zzlGetScore(sptr);
-            vstr = lpGetValue(eptr,&vlen,&vlong);
-            if (vstr == NULL)
-                ele = sdsfromlonglong(vlong);
-            else
-                ele = sdsnewlen((char*)vstr,vlen);
+        if (eptr != NULL) {
+            sptr = lpNext(zl, eptr);
+            serverAssertWithInfo(NULL, zobj, sptr != NULL);
 
-            node = zslInsert(zs->zsl,score,ele);
-            serverAssert(dictAdd(zs->dict,ele,&node->score) == DICT_OK);
-            zzlNext(zl,&eptr,&sptr);
+            while (eptr != NULL) {
+                score = zzlGetScore(sptr);
+                vstr = lpGetValue(eptr, &vlen, &vlong);
+                if (vstr == NULL)
+                    ele = sdsfromlonglong(vlong);
+                else
+                    ele = sdsnewlen((char *) vstr, vlen);
+
+                node = zslInsert(zs->zsl, score, ele);
+                serverAssert(dictAdd(zs->dict, ele, &node->score) == DICT_OK);
+                zzlNext(zl, &eptr, &sptr);
+            }
```

Comment:
maybe we can easily avoid indenting all these lines by adding an `else` like
```c
        if (eptr != NULL) {
            sptr = lpNext(zl, eptr);
            serverAssertWithInfo(NULL, zobj, sptr != NULL);
        } else {
            eptr = NULL;
        }

or actually, just set it to NULL when it is declared.

Reply:
Assuming you meant `sptr` in the else branch, I don't think we need that assignment at all since it is never accessed if `eptr == NULL`, and if eptr isn't NULL sptr is guaranteed to be not NULL as well.